elastic/eui
# [EuiBasicTable][EuiInMemoryTable][EuiDataGrid] Update to use `EuiTablePagination` component defaults

## Summary

This PR is part 2 of 2 of the work required to allow all EUI tables and grids to have their pagination page size configurable by consumer defaults (see https://github.com/elastic/kibana/issues/56406 for more context).

> ⚠️  Note that there is still actual work that needs to be done within Kibana itself to take advantage of the new component defaults functionality:
> 1. The top-level `<EuiProvider>` in each Kibana app needs to be configured with user-preferred `itemsPerPage` and/or `itemsPerPageOptions`
> 2. Every single instance of `EuiBasicTable`, `EuiInMemoryTable`, and `EuiDataGrid` needs to be configured to not initially pass a defined `pagination.pageSize`/`pagination.pageSizeOptions`. See https://github.com/elastic/eui/commit/ab3016d4423a2ef24415634aff7f723a1ce42274 for an example of this change

This PR is also the last PR in the `provider/component-defaults` feature branch - once this PR merges in, the final feature PR itself will be opened up against main 🚢

As always, I strongly recommend [following along by commit](https://github.com/elastic/eui/pull/6993/commits) because there's a lot of RTL test tech debt as well as docs changes/tweaks in here.

## QA

- `gh pr checkout 6993`
- `yarn && yarn start`
- Go to http://localhost:8030/#/tabular-content/data-grid and http://localhost:8030/#/tabular-content/in-memory-tables
- Confirm the first tables/grids default to a page size of 10, and when clicked, show `10, 25, 50` page sizes
- Open `src-docs/src/views/app_context.js`
- On the `<EuiProvider` component, add `componentDefaults={{ EuiTablePagination: { itemsPerPage: 20, itemsPerPageOptions: [10, 20, 0] }}}`
- Save and refresh the above pages
- [x] Confirm the first tables/grids on the page now default to a page size of 20 and show `10, 20, show all rows` options

### General checklist

~- [ ] Added **[documentation](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/documenting)**~

~- [ ] Checked in both **light and dark** modes~
~- [ ] Checked in **mobile**~
~- [ ] Checked in **Chrome**, **Safari**, **Edge**, and **Firefox**~
~- [ ] Checked for **accessibility** including keyboard-only and screenreader modes~
~- [ ] Checked **[Code Sandbox](https://codesandbox.io/)** works for any docs examples~
~- [ ] Updated the **[Figma](https://www.figma.com/community/file/964536385682658129)** library counterpart~
